test/net/sourceforge/kolmafia/maximizer/MaximizerTest.java

```diff
@@ -175,27 +173,113 @@ public void keepsBonusBjornUnchanged() {
     assertEquals(25, modFor("Meat Drop"), 0.01);
   }
 
+  @Test
+  public void maxKeywordStopsCountingBeyondTarget() {
+    addItem("hardened slime hat");
+    addItem("bounty-hunting helmet");
+    addSkill("Refusal to Freeze");
+    setStats(0, 0, 200);
+    assertTrue(maximize("cold res 3 max, 0.1 item drop"));
+
+    assertEquals(3, modFor("Cold Resistance"), 0.01);
+    assertEquals(20, modFor("Item Drop"), 0.01);
+
+    recommendedSlotIs(EquipmentManager.HAT, "bounty-hunting helmet");
+  }
+
+  @Test
+  public void startingMaxKeywordTerminatesEarlyIfConditionMet() {
+    addItem("hardened slime hat");
+    addItem("bounty-hunting helmet");
+    addSkill("Refusal to Freeze");
+    setStats(0, 0, 200);
+    maximize("3 max, cold res");
+
+    assertTrue(
+        Maximizer.boosts.stream()
+            .anyMatch(
+                b -> b.toString().contains("(maximum achieved, no further combinations checked)")));
+  }
+
+  @Test
+  public void minKeywordFailsMaximizationIfNotHit() {
+    addItem("helmet turtle");
+    assertFalse(maximize("mus 2 min"));
+    // still provides equipment
+    assertEquals(1, modFor("Buffed Muscle"), 0.01);
+  }
+
+  @Test
+  public void clownosityTriesClownEquipment() {
+    addItem("clown wig");
+    setStats(0, 0, 10);
+    assertFalse(maximize("clownosity"));
+    // still provides equipment
+    recommendedSlotIs(EquipmentManager.HAT, "clown wig");
+  }
+
+  @Test
+  public void clownositySucceedsWithEnoughEquipment() {
+    addItem("clown wig");
+    addItem("polka-dot bow tie");
+    setStats(0, 10, 10);
+    assertTrue(maximize("clownosity"));
+    recommendedSlotIs(EquipmentManager.HAT, "clown wig");
+    recommendedSlotIs(EquipmentManager.ACCESSORY1, "polka-dot bow tie");
+  }
+
   private void equip(int slot, String item) {
     EquipmentManager.setEquipment(slot, AdventureResult.parseResult(item));
   }
 
   private void addItem(String item) {
-    AdventureResult.addResultToList(KoLConstants.inventory, AdventureResult.parseResult(item));
+    addItem(item, 1);
   }
 
-  private void loadInventory(String jsonInventory) {
-    try {
-      InventoryManager.parseInventory(new JSONObject(jsonInventory));
-    } catch (JSONException e) {
-      fail("Inventory parsing failed.");
+  private void addItem(String item, int count) {
+    for (int i = 0; i < count; i++) {
+      AdventureResult.addResultToList(KoLConstants.inventory, AdventureResult.parseResult(item));
     }
   }
 
+  private void addEffect(String effect) {
+    KoLConstants.activeEffects.add(EffectPool.get(EffectDatabase.getEffectId(effect)));
+  }
+
+  private void addSkill(String skill) {
+    KoLCharacter.addAvailableSkill(skill);
+  }
+
+  private void setStats(int muscle, int mysticality, int moxie) {
```

Comment:
(I generally dislike functions with multiple parameters of the same type, but in this case, there's a canonical + well-known order.)